tutorials/scripting/evaluating_expressions.rst

```diff
@@ -0,0 +1,208 @@
+|                        | If at least one of them is a floating-point number, returns a floating-point value. |
++------------------------+-------------------------------------------------------------------------------------+
+| Modulo (``%``)         | Returns the remainder of an integer division.                                       |
++------------------------+-------------------------------------------------------------------------------------+
+
+Spaces around operators are optional. Also, keep in mind the usual
+`order of operations <https://en.wikipedia.org/wiki/Order_of_operations>`__
+applies. Use parentheses to override the order of operations if needed.
+
+All the Variant types supported in Godot can be used: integers, floating-point
+numbers, strings, arrays, dictionaries, colors, vectors, …
+
+Arrays and dictionaries can be indexed like in GDScript::
+
+    # Returns 1.
+    [1, 2][0]
+
+    # Returns 3. Negative indices can be used to count from the end of the array.
+    [1, 3][-1]
+
+    # Returns "green".
+    {"favorite_color": "green"}["favorite_color"]
+
+    # All 3 lines below return 7.0 (Vector3 is floating-point).
+    Vector3(5, 6, 7)[2]
+    Vector3(5, 6, 7)["z"]
+    Vector3(5, 6, 7).z
+
+Passing variables to an expression
+----------------------------------
+
+In addition to, you can pass variables to an expression. These variables will then
+become available in the expression's "context" and will be substituted when used
+in the expression::
+
+    var expression = Expression.new()
+    # Define the variable names first in the second parameter of `parse()`.
+    expression.parse("20 + 2 * x", ["x"])
+    # Then define the variable values in the first parameter of `execute()`.
+    var result = expression.execute([5])
+    print(result)  # 30
+
+Both the variable names and variable values **must** be specified as an array,
+even if you only define one variable.
+
+Setting a base instance for the expression
+------------------------------------------
+
+By default, an expression has a base instance of ``null``. This means the
+expression has no base instance associated to it.
+
+When calling :ref:`Expression.execute() <class_Expression_method_execute>`,
+you can set the value of the ``base_instance`` parameter to a specific object
+instance such as ``self``, another script instance or even a singleton::
+
+    func double(number):
+        return number * 2
+
+
+    func _ready():
+        var expression = Expression.new()
+        expression.parse("double(10)")
+
+        # This won't work, since we're not passing the current script as the base instance.
+        var result = expression.execute([], null)
+        print(result)  # null
+
+        # This will work won't work, since we're not passing the current script
+        # as the base instance.
+        result = expression.execute([], self)
+        print(result)  # 20
+
+    print(result)  # 30
```

Comment:
Where did print(result) # 30 come from?  it looks like it's outside of any function not sure what happened here.